{{ cookiecutter.repo_name }}/docs/user_guide/loading_environment_variables.md

```diff
@@ -0,0 +1,77 @@
+# Loading environment variables
+
+We use [`direnv`][direnv] to load environment variables, as it ensures you only have project-specific variables loaded
+_when you are inside the project_, otherwise these variables are not loaded. This can prevents accidental conflicts
+with identically named variables.
+
+```{contents}
+:local:
+:depth: 2
+```
+
+## Using `direnv`
+
+To load the environment variables, first [install `direnv`](#installing-direnv){% if cookiecutter.create_secrets_file == "Yes" %}, and make sure you have a `.secrets` file to [store secrets and credentials](#storing-secrets-and-credentials){% endif %}; then:
+
+1. Open your terminal;
+2. Navigate to the project folder; and
+   - You should see the following message:
+     ```shell
+     direnv: error .envrc is blocked. Run `direnv allow` to approve its content.
+     ```
+3. Allow `direnv`.
+   ```shell
+   direnv allow
+   ```
+
+You only need to do this **once**, and again each time `.envrc` {% if cookiecutter.create_secrets_file == "No" %}is{% else %}and `.secrets` are{% endif %} modified.
+
+### Installing `direnv`
+
+These instructions assume you are running on macOS with administrator privileges using a bash terminal. For other ways
+of installing `direnv`, and its shell hooks, consult the package's [documentation][direnv].
+
+1. Open your terminal;
+2. Install `direnv` via [Homebrew][homebrew];
+   ```shell
+   brew install direnv
+   ```
+3. Add the shell hooks to your `.bash_profile`;
+   ```shell
+   echo 'eval "$(direnv hook bash)"' >> ~/.bash_profile
+   ```
+4. Check that the shell hooks have been added correctly; and
+   ```shell
+   cat ~/.bash_profile
+   ```
+   - This should display `eval "$(direnv hook bash)"`
+5. Restart your terminal.
+{% if cookiecutter.create_secrets_file == "Yes" %}
+## Storing secrets and credentials
+
+Secrets and credentials must be stored in the `.secrets` file. **This file is not version-controlled**, so no secrets
+should be committed to {{ cookiecutter.repository_hosting_platform }}.
+
+In your terminal navigate to the root folder, and create a `.secrets` file.
+
+```shell
+touch .secrets
+```
```

Comment:
I am not quite sure why they need to create a `.secrets` file given that this section is only displayed if `% if cookiecutter.create_secrets_file == "Yes" %` which automatically creates a .secrets file in the root directory.

They need to create a `secrets` file if they had opted `cookiecutter.create_secrets_file == "No"` and then changed their mind. Or am I missing something?

Reply:
Ah yes, the **creator** of the project wouldn't need to do it. However, anyone subsequently cloning the project would need to create their own `.secrets` file.

Reply:
Ah ok, so maybe we should specify that? It is not very clear whom these instructions are aimed at, and I think both the creator and the collaborator of the project will need these type of guidance. 

Also, there is the additional possibility that a creator will add .secrets files later on in a project. We may want to account for this option to?

Reply:
That's a really good point - do you think I should remove all the `.secrets` Jinja conditionals from this file so that anyone wanting to create a secrets file in the future can do so? How do you think we should handle other parts of the documentation that might be missing secrets file information (e.g. the structure docs)? Or should we **always** create a secrets file, and leave it up to the users to decide if they want to use it or not?

To cover both creators and collaborators, would adding some lines such as "if relevant", "if missing" etc. cover it do you think?

Reply:
Have discussed this with @exfalsoquodlibet offline. We both agreed that secrets should be mandatory (and not optional). We'll leave this for another PR.

Marking this as **Resolved**.